lib/XTR/ReportScheduleTask.php

```diff
@@ -126,51 +126,63 @@ private function runReportSchedule()
                 $rs = $this->reportScheduleFactory->getById($reportSchedule->reportScheduleId);
                 $rs->previousRunDt = $rs->lastRunDt;
                 $rs->lastRunDt = time();
+                $rs->isActive = 0;
                 $rs->save();
 
-                // Get the generated saved as report name
-                $saveAs = $this->reportService->generateSavedReportName($reportSchedule->reportName, $reportSchedule->filterCriteria);
-
                 $this->log->debug('Last run date is updated to '. $rs->lastRunDt);
 
-                // Run the report to get results
-                $result =  $this->reportService->runReport($reportSchedule->reportName, $reportSchedule->filterCriteria, $reportSchedule->userId);
-                $this->log->debug('Run report results: %s.', json_encode($result, JSON_PRETTY_PRINT));
+                try {
+                    // Get the generated saved as report name
+                    $saveAs = $this->reportService->generateSavedReportName($reportSchedule->reportName, $reportSchedule->filterCriteria);
+
+
+                    // Run the report to get results
+                    $result =  $this->reportService->runReport($reportSchedule->reportName, $reportSchedule->filterCriteria, $reportSchedule->userId);
+                    $this->log->debug(__('Run report results: %s.', json_encode($result, JSON_PRETTY_PRINT)));
+
+                    //  Save the result in a json file
+                    $fileName = tempnam(sys_get_temp_dir(), 'reportschedule');
```

Comment:
I know this hasn't been changed here, but I didn't notice this before.
We need to always write temporary files into the library temp folder and not the FS folder please.